scripts/getData.py

```diff
@@ -1,135 +1,99 @@
-	# Replace html
-	cleanString = h.unescape( string )
-	# Replace unicode
-	cleanString2 = unicodedata.normalize( 'NFKD', cleanString ).encode( 'ascii','ignore' )
-	# Capitalize string
-	return cleanString2[0].capitalize() + cleanString2[1:]
-
+def appendData(cafeId, cafeName, cafeLoc):
+	print(cafeId + ") " + cafeName + " in " + cafeLoc)
+	return {'id':cafeId, 'label':cafeName, 'desc':cafeLoc}
 
-def cleanUpLocation( stringToClean ):
+def clean(stringToClean):
 	# Remove beginning and ending whitespace
 	string = stringToClean.strip()
 	# Replace html
-	cleanString = h.unescape( string )
+	cleanString = h.unescape(string)
 	# Replace unicode
-	cleanString2 = unicodedata.normalize( 'NFKD', cleanString ).encode( 'ascii','ignore' )
-
-	# Handle city/state (length will be longer than 2 chars)
-	if len( cleanString2 ) > 2:
-		# Capitalize each first letter
-		string2 = cleanString2.title()
-	else:
-		string2 = cleanString2[0].capitalize() + cleanString2[1:]
-
-	return string2
+	return unicodedata.normalize('NFKD', cleanString)
 
 
 # Finds the cafeteria id and name
-def getBonAppMenuData( url, id ):
-	# Construct the full url
-	url2 = ''.join( [url, str(id)] )
-
-   	# Receive the content of url and convert it to a string
-	response  = urlopen( url2 )
-	data 	  = str( response.read() )
+def getBonAppMenuData(url, id):
+	# Our constructed JSON data
+	response = requests.get(url, params={'cafe': id})
+	data = response.json()
 
 	try:
 		# Grab cafe Id from JSON
-		cafeId    = json.loads( data )["cafes"].keys()[0]
+		cafeId = list(data["cafes"].keys())[0]
 
 		# Grab cafe name from JSON
-		cafeName  = json.loads( data )["cafes"][cafeId]["name"]
+		cafeName = data["cafes"][cafeId]["name"]
 
 
 		# We want to display titles at least. We can do without location.
-		try:
-			# Grab cafe city from JSON
-			cafeCity  = json.loads( data )["cafes"][cafeId]["city"]
-		except:
-			cafeCity  = ""
-		try:
-			# Grab cafe state from JSON
-			cafeState = json.loads( data )["cafes"][cafeId]["state"]
-		except:
-			cafeState = ""
+		# Grab cafe city from JSON
+		cafeCity = data["cafes"][cafeId].get("city", "")
 
+		# Grab cafe state from JSON
+		cafeState = data["cafes"][cafeId].get("state", "")
 
 		# Formatting city and state strings
 		# Both empty
 		if cafeCity == "" and cafeState == "":
-			cafeLoc	= "No location listed"
+			cafeLoc = "No location listed"
 		# Only city
 		elif cafeCity != "" and cafeState == "":
-			cafeLoc = cleanUpLocation( cafeCity )
+			cafeLoc = clean(cafeCity)
 		# Only State
 		elif cafeCity == "" and cafeState != "":
-			cafeLoc = cleanUpLocation( cafeState )
+			cafeLoc = clean(cafeState)
 		# City and State
 		else:
-			cafeLoc = cleanUpLocation( cafeCity ) + ", " + cleanUpLocation( cafeState )
-
+			cafeLoc = clean(cafeCity) + ", " + clean(cafeState)
 
 		# Clean up the cafe name
-		cafeName  = cleanUpTitle( cafeName )
+		cafeName = clean(cafeName)
 
 		# Construct the full return string
-		appendData( cafeId, cafeName, cafeLoc )
```

Comment:
Could you just inline that function here? It's only called once and it's not named appropriately for what it does IMO.

Reply:
Done 👍